README.md

```diff
@@ -9,6 +9,83 @@ Installation
 go get github.com/segmentio/stats
 ```
 
+Migration to v4
+---------------
+
+Version 4 of the stats package introduced a new way of producing metrics based
+on defining struct types with tags on certain fields that define how to interpret
+the values. This approach allows for much more efficient metric production as it
+allows the program to do quick assignments and increments of the struct fields to
+set the values to be reported, and submit them all with one call to the stats
+engine, resulting in orders of magnitude faster metrics production. Here's an
+example:
+```go
+type funcMetrics struct {
+    calls struct {
+        count int           `metric:"count" type:"counter"`
+        time  time.Duration `metric:"time"  type:"histogram"`
+    } `metric:"func.calls"`
+}
+```
+```go
+t := time.Now()
+f()
+callTime := time.Now().Sub(t)
+
+m := funcMetrics{}
+m.calls.count = 1
+m.calls.time = callTime
+
+// Prior to v4 this could have been:
+//
+//   stats.Incr("func.calls.count")
+//   stats.ObserveDuration("func.calls.time", callTime)
+//
+stats.Report(m)
```

Comment:
I'm less excited by this new API. Depending on whether `funcMetrics` is reused or not, it adds 8 extra lines of code. Though in the case it is reused, it could offer more consistency.